code/modules/projectiles/updated_projectiles/guns/flamer.dm

```diff
@@ -515,10 +500,14 @@
 			show_message(text("Your suit protects you from most of the flames."), 1)
 			return CLAMP(. * 1.5, 0.75, 1) //Min 75% resist, max 100%
 
+/mob/living/carbon/Xenomorph/run_armor_check(def_zone = null, attack_flag = "melee")
+	if(attack_flag == "fire" && (xeno_caste.caste_flags & CASTE_FIRE_IMMUNE))
+		return 1
+	. = ..()
```

Comment:
```suggestion
	return ..()
```
Though there's nothing to return yet, as `getarmor()` is not yet defined for xenos.